### PR TITLE
WELD-2387 Configurator defining a custom bean with no callback will f…

### PR DESCRIPTION
**Pull request by manovotn**
…ail with ISE during deployment.

### CODE REVIEW

File: impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/BeanConfiguratorImpl.java [hunk 1]
```diff
@@ -311,7 +311,12 @@ public BeanConfiguratorImpl(Class<?> defaultBeanClass, BeanDeploymentFinder bean
     }
 
     public Bean<T> complete() {
-        return new ImmutableBean<>(this);
+        ImmutableBean<T> result = new ImmutableBean<>(this);
+        if (createCallback == null) {
```
Comment (manovotn) [inline impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/BeanConfiguratorImpl.java]: I am aware that creating bean and only then throwing exception isn't the best way, but without having `ImmutableBean`, there is no way to access information like types of bean or qualifiers.

**Comment by manovotn** [comment]
The failure is TCK-aspected - there is a [test](https://github.com/cdi-spec/cdi-tck/blob/master/impl/src/main/java/org/jboss/cdi/tck/tests/extensions/configurators/bean/LifecycleObserver.java#L165) creating a bean with no callback (but the testcase is to verify defaults scope, so we can fix that - already on it).

File: impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/BeanConfiguratorImpl.java (hunk 1 above)
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/BeanConfiguratorImpl.java]: Well, you could make `BeanAttributesConfiguratorImpl.qualifiers` and `types` package-private and use something like:

````
throw BeanLogger.LOG.noCallbackSpecifiedForCustomBean("Configurator [types: " + Formats.formatTypes(attributes.types) + ", qualifiers: " + Formats.formatAnnotations(attributes.qualifiers) + "]");
````

Hm, `BeanAttributesConfiguratorImpl.types` should be final - pls add this change too.

**Comment by mkouba** [comment]
Please also update the exclude list: `jboss-tck-runner/src/test/tck20/tck-tests.xml`.

**Pull request merged**